downloader.ts

```diff
@@ -0,0 +1,64 @@
+import * as fs from 'fs';
+import { https } from 'follow-redirects';
+import MemoryStream from 'memorystream';
+import { keccak256 } from 'js-sha3';
+
+function getVersionList () {
+  console.log('Retrieving available version list...');
+
+  return new Promise<string>((resolve, reject) => {
+    const mem = new MemoryStream(null, { readable: false });
+    https.get('https://binaries.soliditylang.org/bin/list.json', function (response) {
+      if (response.statusCode !== 200) {
```

Comment:
It's something that has already been here before but we should change this so that anything between 200 and 299 is a success. I think we also have the same check in other places in the code

But I'd do it in a separate PR since it's a functional change, not just a refactor and is also unrelated to tests.

Reply:
`response.ok`?

Reply:
I don't think the `https` nodejs module has such `ok` property. I know that the [fecth API](https://developer.mozilla.org/en-US/docs/Web/API/Response) has it, but it is not what we are using here. However, we could mimic its behavior.

Reply:
Ah, you're right, I was actually looking at the fetch API when I made the comment.